code/modules/urist/antagonist/revenant/archetypes/chimera.dm

```diff
@@ -101,51 +101,103 @@
+				to_chat(H, SPAN_WARNING("By reviving, you have significantly violated the laws of reality!"))
+
+				// Short-circuit to a return; because reviving is more distortey
+				if(valid_revenant)
+					revenant.total_distortion += BSR_DISTORTION_GROWTH_OVER_SECONDS(1500, BSR_DEFAULT_DISTORTION_PER_TICK, BSR_DEFAULT_DECISECONDS_PER_TICK)
+					return TRUE
 
 		for(var/obj/item/organ/external/E in H.bad_external_organs)
 			if(!pending)
 				break
 
-			if((E.status & ORGAN_ARTERY_CUT))
-				to_chat(H, SPAN_NOTICE("You mend the torn artery in your [E.name], stemming the worst of the bleeding."))
-				E.status &= ~ORGAN_ARTERY_CUT
-				pending = FALSE
-				break
+			if(prob(70))
+				continue
 
-			for(var/datum/wound/W in E.wounds)
-				if(W.bleeding())
-					W.bleed_timer = 0
-					E.status &= ~ORGAN_BLEEDING
-					to_chat(H, SPAN_NOTICE("You knit together severed veins, stemming the bleeding from your [E.name]."))
-					pending = FALSE
-					break
+			E.rejuvenate()
+			to_chat(H, SPAN_NOTICE("Your [E.name] itches as you knit its wounds shut."))
+			pending = FALSE
+			break
 
-			if(E.damage >= 5)
-				to_chat(H, SPAN_NOTICE("Your [E.name] itches as you knit its wounds shut."))
+		if(pending && prob(20))
+			if(H.should_have_organ(BP_HEART) && H.vessel.total_volume < H.species.blood_volume)
+				to_chat(H, SPAN_NOTICE("Your veins seem to swell back with unnaturally restored blood."))
+				H.vessel.add_reagent(/datum/reagent/blood, H.species.blood_volume - H.vessel.total_volume)
+				// Not using restore_blood() because it would double-check the same vars we already checked
+				// H.restore_blood()
 				pending = FALSE
-				break
 
-	if(!pending)
-		var/datum/bluespace_revenant/revenant = src?.mind?.bluespace_revenant
-		if(istype(revenant))
-			revenant.total_distortion += BSR_DISTORTION_GROWTH_OVER_SECONDS(30, BSR_DEFAULT_DISTORTION_PER_TICK, BSR_DEFAULT_DECISECONDS_PER_TICK)
+	var/mob/living/L = src
+
+	if(istype(L))
+		// Brain
+		if(pending)
+			var/brain_dmg = L.getBrainLoss()
+			if(brain_dmg)
+				L.adjustBrainLoss(-brain_dmg)
+				to_chat(L, SPAN_NOTICE("Your brain itches as you knit its wounds shut."))
+
+				if(valid_revenant)
+					revenant.total_distortion += BSR_DISTORTION_GROWTH_OVER_SECONDS(brain_dmg*3, BSR_DEFAULT_DISTORTION_PER_TICK, BSR_DEFAULT_DECISECONDS_PER_TICK)
+					return FALSE
+
+		// Clone
+		if(pending)
+			var/clone_dmg = L.getCloneLoss()
+			if(clone_dmg)
+				L.adjustCloneLoss(-clone_dmg)
+				to_chat(L, SPAN_NOTICE("You feel your genome rearranging. Somehow."))
+
+				if(valid_revenant)
+					revenant.total_distortion += BSR_DISTORTION_GROWTH_OVER_SECONDS(clone_dmg, BSR_DEFAULT_DISTORTION_PER_TICK, BSR_DEFAULT_DECISECONDS_PER_TICK)
+					return FALSE
+
+		// Tox
+		if(pending)
+			var/tox_dmg = L.getToxLoss()
+			if(tox_dmg)
+				L.adjustToxLoss(-tox_dmg)
+				to_chat(L, SPAN_NOTICE("You purge absorbed toxins from your cells (though not your bloodstream)!"))
```

Comment:
For humans toxloss is just damage to organs (excluding the brain) nowadays.

Reply:
Yeah, but I'm accounting for **non**humanoids, potentially.

Reply:
Besides maybe the phoron pistol I'm not sure there's many instances of simple mobs or any other living mobs taking toxin damage. It's just going to be a kind of strange message for most circumstances.

Reply:
Adjusted